wallet/address.go

```diff
@@ -0,0 +1,59 @@
+package wallet
+
+import (
+	"encoding/hex"
+	"fmt"
+	"perun.network/go-perun/wallet"
+)
+
+// PubKeyLength is the length of a Cardano public key in bytes
+const PubKeyLength = 32
+
+// PubKey represents a Cardano public key struct (Ledger.Crypto.PubKey in the plutus-ledger library).
+// The KeyString is a hex string (without "0x") representing the bytes of the public key
+type PubKey struct {
+	KeyString string `json:"getPubKey"`
+}
+
+// MarshalBinary decodes the hexadecimal key string and returns the represented bytes after verifying the PubKeyLength
```

Comment:
Well, `MarshalBinary` and `UnmarshalBinary` *are* supposed to be the "smart constructors for `PubKey`. I don't really see any gains in adding two functins that perform the checks for marshalling and unmarshalling, especially since `UnmarshalBinary` operates on a pointer receiver (I don't see atm how a "smart constructor" would work then - despite adding a function that just checks the length) ?